### ISSUE TITLE
Can’t cast a single hashtable to List of custom objects

### ISSUE THREAD

**Issue by Yaekiou** [PowerShell/PowerShell#24557]
### Prerequisites

- [X] Write a descriptive title.
- [X] Make sure you are able to repro it on the [latest released version](https://github.com/PowerShell/PowerShell/releases)
- [X] Search the existing issues.
- [X] Refer to the [FAQ](https://github.com/PowerShell/PowerShell/blob/master/docs/FAQ.md).
- [X] Refer to [Differences between Windows PowerShell 5.1 and PowerShell](https://learn.microsoft.com/powershell/scripting/whats-new/differences-from-windows-powershell).

### Steps to reproduce

Hi, A single hashtable object can be cast to `System.Array` of custom objects

```powershell
using namespace System.Collections.Generic

class Foo {
  [int]$A
}

# OK
[Foo[]]@{ A = 1 }

# OK
[List[Foo]]@(@{ A = 2 })

# OK
[List[Foo]][Foo]@{ A = 3 }
```

But can’t cast to `List<T>`
```powershell
# Error
[List[Foo]]@{ A = 4 }
```

Is this a bug?

### Expected behavior

```console
PS> [List[Foo]]@{ A = 4 }
A
-
4
```

### Actual behavior

```console
PS> [List[Foo]]@{ A = 4 }
InvalidArgument: ……
```

### Environment data

```powershell
PS> $PSVersionTable
Name
Value
--—-
- ----
PSVersion
7.4.6
PSEdition
Core
GitCommitId
7.4.6
OS
Ubuntu 20.04.6 LTS
Platform
Unix
PSCompatibleVersions
{1.0, 2.0,
3.0, 4.0...}
PSRemotingProtocolVersion
2.3
SerializationVersion
1.1.0.1
WSManStackVersion
3.0
```

### Visuals

![IMG_0241](https://github.com/user-attachments/assets/55615abb-4cab-4dbc-9905-84e3ee762abe)

```powershell
$PSVersionTable
```
![IMG_0242](https://github.com/user-attachments/assets/6c153628-f2d0-48e7-821e-5fba56229177)

**Comment by mklement0** [comment]
Good point; a variant of this problem was previously reported in the context of `System.Collections.ArrayList`:

* #11749
